src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -4,12 +4,21 @@
 using System;
 using System.Diagnostics;
 using System.Numerics;
+using System.Runtime.CompilerServices;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
 {
     public struct MemoryPoolIterator
     {
+        private const ulong _xorPowerOfTwoToHighByte = (0x07ul       |
+                                                        0x06ul <<  8 |
+                                                        0x05ul << 16 |
+                                                        0x04ul << 24 |
+                                                        0x03ul << 32 |
+                                                        0x02ul << 40 |
+                                                        0x01ul << 48 ) + 1;
```

Comment:
We are doing a mathematical transform that doesn't care about the underlying byte order. Issue with using multiple bytes in hex directly is it could be interpreted in two ways; however the shifts are just changing its mathematical number independent of underlying byte order.

The compiler changes this to 
```csharp
xorPowerOfTwoToHighByte = unsigned int64(283686952306184) // 0x0001020304050608
```